test/metric_value_benchmark/container_insights_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type ContainerInsightsTestRunner struct {
+	ECSBaseTestRunner
+}
+
+var _ IECSTestRunner = (*ContainerInsightsTestRunner)(nil)
+
+func (t *ContainerInsightsTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := t.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = t.validateContainerInsightsMetrics(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *ContainerInsightsTestRunner) getTestName() string {
+	return "ContainerInstance"
+}
+
+func (t *ContainerInsightsTestRunner) getAgentConfigFileName() string {
+	return "./agent_configs/container_insights.json"
+}
+
+func (t *ContainerInsightsTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
+}
+
+func (t *ContainerInsightsTestRunner) getMeasuredMetrics() []string {
+	return []string{
+		"instance_memory_utilization", "instance_number_of_running_tasks", "instance_memory_reserved_capacity",
```

Comment:
Would it be better to leave comments here with the public document that talks about metrics that CWAgent is gathering

Reply:
Imo this is the source for public docs, not the other way around. I don't mind it but why do you think that will be helpful?

Reply:
It would be better for new dev to know what we support for customers currently. So if the values are mismatch, we can instantly fixing it.

Reply:
Fixing it here doesn't fix it in the agent. This is just testing.